lib/Conversion/TorchToStablehlo/Uncategorized.cpp

```diff
@@ -0,0 +1,311 @@
+
+  LogicalResult
+  matchAndRewrite(Aten_MakePerTensorQuantizedTensorOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    auto opUser = *op.getResult().user_begin();
+    auto users = op.getResult().getUsers();
+    if (!(std::distance(users.begin(), users.end()) == 1) ||
+        !isa<AtenDequantizeSelfOp>(opUser)) {
+      return failure();
+    }
+    // [TODO] verify that zeroPoint and Scale matches with the input operand
+    // type.
+    RankedTensorType outputType = cast<RankedTensorType>(
+        getTypeConverter()->convertType(opUser->getResult(0).getType()));
+
+    rewriter.replaceOpWithNewOp<stablehlo::UniformDequantizeOp>(
+        opUser, outputType, adaptor.getOperands().front());
+
+    rewriter.eraseOp(op);
+    return success();
+  }
+};
+} // namespace
+
+namespace {
+class ConvertAtenQuantizePerChannelOp
+    : public OpConversionPattern<AtenQuantizePerChannelOp> {
+public:
+  using OpConversionPattern::OpConversionPattern;
+
+  LogicalResult
+  matchAndRewrite(AtenQuantizePerChannelOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    auto *zeroPoints = op.getZeroPoints().getDefiningOp();
+    if (!zeroPoints || !isa<ValueTensorLiteralOp>(zeroPoints)) {
+      return failure();
+    }
+    auto zeroPointsOp = mlir::cast<ValueTensorLiteralOp>(zeroPoints);
+
+    llvm::SmallVector<int64_t, 4> zeroPointsVec;
+    for (auto zp : zeroPointsOp.getValue().getValues<llvm::APInt>()) {
+      zeroPointsVec.emplace_back(zp.getSExtValue());
+    }
+
+    auto scales = op.getScales().getDefiningOp();
+    if (!scales || !isa<ValueTensorLiteralOp>(scales)) {
+      return failure();
+    }
+
+    llvm::SmallVector<double, 4> scalesVec;
+    auto scalesOp = mlir::cast<ValueTensorLiteralOp>(scales);
+    for (auto scale : scalesOp.getValue().getValues<llvm::APFloat>()) {
+      scalesVec.emplace_back(scale.convertToDouble());
+    }
+
+    auto axis = op.getAxis().getDefiningOp();
+    if (!axis || !isa<ConstantIntOp>(axis)) {
+      return failure();
+    }
+    auto axisOp = mlir::cast<ConstantIntOp>(axis);
+    auto axisValue = axisOp.getValueAttr().getInt();
+
+    auto users = op.getResult().getUsers();
+    auto opUser = *op.getResult().user_begin();
+    if (!(std::distance(users.begin(), users.end()) == 1) ||
+        !isa<AtenIntReprOp>(opUser)) {
+      return failure();
+    }
+
+    auto inputElemType =
+        mlir::cast<RankedTensorType>(
+            getTypeConverter()->convertType(op.getOperands().front().getType()))
+            .getElementType();
+
+    mlir::Type dtype =
+        cast<ValueTensorType>(op->getResult(0).getType()).getDtype();
+    int32_t bitWidth = 0;
+    int32_t flags = quant::QuantizationFlags::FlagValue::Signed;
+    if (isa<QUInt8Type>(dtype)) {
+      bitWidth = 8;
+      flags = 0;
+    } else if (isa<QInt8Type>(dtype) || isa<QUInt8Type>(dtype)) {
```

Comment:
```suggestion
    } else if (isa<QInt8Type>(dtype)) {
```
`QUInt8Type` already been checked in the previous branch

Reply:
Fixed. thanks for pointing it out.